framework/src/main/java/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -968,6 +1004,40 @@ public boolean isSubtype(AnnotationMirror subAnno, AnnotationMirror superAnno) {
                     return false;
             }
         }
+
+        /**
+         * Return true if the target annotation is in the original value qualifier hierarchy.
+         *
+         * @param anno the annotation to check
+         * @return true if the target annotation is in the original value qualifier hierarchy
+         */
+        private boolean inValueQualifierHierarchy(AnnotationMirror anno) {
+            return AnnotationUtils.areSameByName(anno, ARRAYLEN_NAME)
```

Comment:
Can you think about extracting the set of supported qualifiers in `createSupportedTypeQualifiers` above and then here checking for containment?